src/main/java/duke/Parser.java

```
@@ -0,0 +1,195 @@
+                    taskList.deleteTask(taskList.getTask(index));
+                } catch (NumberFormatException e) {
+                    Ui.showError("Invalid task number");
+                }
+            } else {
+                // Invalid command format
+                Ui.showError("Invalid command format");
+            }
+
+        } else if (userInput.startsWith("find")) {
+            // IF user wants to find a task
+            String keyword = userInput.substring(5).trim();
+            taskList.findTasks(keyword);
+
+        } else {
+            // Other inputs
+            Ui.showError("I'm sorry, but I don't know what that means");
+        }
+    }
+
+    /**
+     * Parses a string representation of a task and creates a Task object.
+     * This method takes a string representation of a task in a specific format
+     * and parses it to create a corresponding Task object. The input string is
+     * split using the "|" delimiter and the task type, status, description, and
+     * other relevant information are extracted to create the appropriate task.
+     * The created task object is then returned.
+     *
+     * @param line The string representation of a task to be parsed.
+     * @return A Task object created from the parsed string representation.
+     */
+    public static Task parseFileInput(String line) {
+        String[] parts = line.split("\\|");
+        String type = parts[0].trim();
+        boolean isDone = parts[1].trim().equals("1");
+        String description = parts[2].trim();
+
+        Task task = null;
+
+        switch (type) {
+        case "T":
+            task = new ToDos(description);
+            break;
+        case "D":
+            String by = parts[3].trim();
+
+            // Parse the date and time
+            LocalDate deadlineDate = LocalDate.parse(by, DateTimeFormatter.ofPattern("MMM d yyyy"));
+
+            task = new Deadlines(description, deadlineDate);
+            break;
+        case "E":
```

Might be better to add a default branch to ensure all bases are covered? If control flow really should not reach the default branch, it might be better to raise an exception to detect any errors in the program!